tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+
+    sinon.assert.calledWith(dispatchSpy, editUserAccount({
+      errorHandlerId: errorHandler.id,
+      userFields: {
+        biography: '',
+        display_name: user.displayName,
+        homepage: '',
+        location: '',
+        occupation: '',
+        username: user.username,
+      },
+      userId: user.id,
+    }));
+  });
+
+  it('renders a Not Found page when logged user cannot edit another user', () => {
+    const username = 'current-logged-user';
+    const { store } = dispatchSignInActions({
+      userProps: defaultUserProps({
+        username,
+        permissions: [],
+      }),
+    });
+
+    // Create a user with another username.
+    const user = createUserAccountResponse({ username: 'willdurand' });
+    store.dispatch(loadUserAccount({ user }));
+
+    // Try to edit this user with another username.
+    const params = { username: user.username };
+    const root = renderUserProfileEdit({ params, store });
+
+    expect(root.find(NotFound)).toHaveLength(1);
+  });
+
+  it('allows to edit another user if logged user has USERS_EDIT permission', () => {
+    const username = 'current-logged-user';
+    const { store } = dispatchSignInActions({
+      userProps: defaultUserProps({
+        username,
+        permissions: [USERS_EDIT],
+      }),
+    });
+
+    // Create a user with another username.
+    const user = createUserAccountResponse({ username: 'willdurand' });
+    store.dispatch(loadUserAccount({ user }));
+
+    // Try to edit this user with another username.
+    const params = { username: user.username };
+    const root = renderUserProfileEdit({ params, store });
+
+    expect(root.find(NotFound)).toHaveLength(0);
+    expect(root.find('.UserProfileEdit-username'))
+      .toHaveProp('defaultValue', user.username);
+
+    const linkItems = root.find('.UserProfileEdit-user-links li');
+    expect(linkItems.at(0).find(Link))
+      .toHaveProp('children', "View user's profile");
+    // We are on the edit page, so we do not use a link for this item.
+    expect(linkItems.at(1).children()).toHaveText("Edit user's profile");
+  });
+
+  it('does not dispatch editUserAccount action when there is no current user', () => {
+    const { store } = dispatchSignInActions();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = mountUserProfileEdit({ errorHandler, store });
+
+    dispatchSpy.reset();
+
+    root.find('.UserProfileEdit-form').simulate('submit', createFakeEvent());
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders errors', () => {
+    const { store } = dispatchSignInActions();
+    const errorHandler = new ErrorHandler({
+      id: 'some-id',
+      dispatch: store.dispatch,
+    });
+    errorHandler.handle(new Error('unexpected error'));
+
+    const root = renderUserProfileEdit({ errorHandler, store });
+
+    expect(root.find(ErrorList)).toHaveLength(1);
```

Comment:
Should we verify the content of the `ErrorList` as well?

Reply:
Do we do that in other test files? I think it is enough because the `ErrorList` component has its own test suite too.

Reply:
Good point, this would be covered by the tests for `ErrorList`.